grep-0015-remove-swig.md

```diff
@@ -27,18 +29,156 @@ anyway.
 ## Copyright / License
 
 This GREP is licensed as CC-BY-ND.
-Copyright 2018 The GNU Radio Foundation.
+Copyright 2020 Free Software Foundation??
 
 ## Motivation
 
 A lot of us have cursed SWIG programming. If we can get rid of it... well,
 wouldn't that be great. But maybe we can't. Let's find out.
 
+Since SWIG takes in the entire public header file, many methods and class members that are not necessarily intended to be exposed through the python API gets automatically bound.  Pybind11 requires deliberate binding of methods, and the binding code does not get regenerated on every compile.  
+
 ## Description
 
-**To be written**. GREP14 (blocktool) would be a help.
+### Block Binding Code Structure
+
+In each gr module, create a `python/bindings` directory
+
+`python/bindings`
+-  `python_bindings.cpp` <-- c++ code to build up the pybind11 module block by block
+   -  alternatively this can be broken up into a per-block cpp file
+-  `generated/` <-- folder to contain automatically generated binding code with [blockname]_python.hpp
```

Comment:
On second thought, I think this is an unnecessary distinction, especially in-tree.  Let's just put all the binding files in the bindings directory for simplicity